lib/hashgraph.cc

```diff
@@ -276,29 +276,22 @@ void Hashgraph::consume_sequence_and_tag(const std::string& seq,
 //
 
 // TODO? Inline in header.
-void
-Hashgraph::
-consume_fasta_and_tag(
-    std:: string const  &filename,
-    unsigned int	      &total_reads, unsigned long long	&n_consumed
+template<typename SeqIO>
```

Comment:
Why do we need to template this? `consume_fasta_and_tag` (and other `*_fasta_*` methods) will always use the the fastx parser no? Or would you call `consume_fasta_` with a BAM parser instead? That seems weird.

In my mind a Fasta and BAM parser both subclass a common `Parser` so we can pass around a pointer/reference to the instance after we create it (at the point where the decision is made).

Help me see where you are going with this :)